src/stdlib_str2num.fypp

```diff
@@ -0,0 +1,569 @@
+                    int_dp1 = int_dp1*10 + val
+                else if(icount<2*maxdpt)then
+                    int_dp2 = int_dp2*10 + val
+                end if
+            else if( val == period ) then
+                pP = i; aux = 0
+            else
+                exit
+            end if
+        end do
+        pE = i ! Fix the exponent position
+        do while( i<=len(s) )
+           val = iachar(s(i:i))-digit_0
+           if( val < 0 .or. val > 9 ) exit
+           i = i + 1
+        end do
+        p = i
+        resp = pE-min(pP,p) ! If no decimal indicator found it is taken as being in the current p position
+        if( resp <= 0 ) resp = resp+1 
+        !----------------------------------------------
+        ! Get exponential
+        sige = 1
+        if( p<len(s) ) then
+            if( any([le,BE,ld,BD]+digit_0==iachar(s(p:p))) ) p = p + 1
+            if( iachar(s(p:p)) == minus_sign+digit_0 ) then
+                sige = -1
+                p = p + 1
+            else if( iachar(s(p:p)) == plus_sign+digit_0 ) then
+                p = p + 1
+            end if
+        end if
+        
+        i_exp = 0
+        do while( p<=len(s) )
+            val = iachar(s(p:p))-digit_0
+            if( val >= 0 .and. val <= 9) then
+                i_exp = i_exp*10_int8 + val ; p = p + 1
+            else
+                exit
+            end if
+        end do
+        
+        exp_aux = nwnb-1+resp-sige*i_exp
+        if( exp_aux>0 .and. exp_aux<=nwnb+nfnb) then
+            if(icount<=maxdpt)then
+                v = sign*int_dp1*expbase(exp_aux)
+            else
+                v = sign*(int_dp1 + int_dp2*fractional_base(maxdpt-1))*expbase(exp_aux-icount+maxdpt)
+            end if
+        else
+            v = sign*(int_dp1 + int_dp2*fractional_base(maxdpt-1))*10._wp**(sige*i_exp-resp+maxdpt+aux)
+        end if
+        stat = 0
+    end subroutine
+#:endif
+    
+    !---------------------------------------------
+    ! Internal Utility functions
+    !---------------------------------------------
+    
+    elemental function mvs2nwsp(s) result(p)
+        !> move string to position of the next non white space character
+        character(*),intent(in) :: s !> character chain
+        integer(int8) :: p !> position
+        !----------------------------------------------
+        p = 1
+        do while( p<len(s) .and. (iachar(s(p:p))==WS.or.iachar(s(p:p))==LF.or.iachar(s(p:p))==CR) ) 
+            p = p + 1
+        end do
+    end function
+    
+    elemental function mvs2wsp(s) result(p)
+        !> move string to position of the next white space character
+        character(*),intent(in) :: s !> character chain
+        integer(int8) :: p !> position
+        !----------------------------------------------
+        p = 1
+        do while( p<len(s) .and. .not.(iachar(s(p:p))==WS.or.iachar(s(p:p))==LF.or.iachar(s(p:p))==CR) ) 
+            p = p + 1
+        end do
+    end function
```

Comment:
I understand these are internal, but full names (e.g. `move_string_to_nonwhitespace` and `move_string_to_whitespace`) would be clearer in places where these are used.

Reply:
Ok, I'll rename those. What about: `shift_to_nonwhite_space` and `shift_to_white_space` ?

Reply:
I like it but white space, when referred to as a character in computer programming, is a single word "whitespace".

Reply:
done